praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/service/PraktikumsstellenService.java

```diff
@@ -149,6 +158,105 @@ public void deleteAusbildungsPraktikumsstelle(UUID praktikumsstellenId) {
+    private void updateStudiumsPraktikumsstelleWithAssignedNwk(UUID id, UpdateStudiumsPraktikumsstelleWithMeldezeitraumAndAssignedNwkDto praktikumsstelleDto) {
+        Optional<StudiumsPraktikumsstelle> studiumsPraktikumsstelleOptional = studiumsPraktikumsstellenRepository.findById(id);
+        if (studiumsPraktikumsstelleOptional.isEmpty()) {
+            throw new ResourceNotFoundException("Praktikumsstelle nicht gefunden.");
+        }
+
+        StudiumsPraktikumsstelle studiumsPraktikumsstelle = studiumsPraktikumsstelleOptional.get();
+
+        //Check if any field which is not supposed to change changed
+        if (studiumsPraktikumsstelle.getReferat() != praktikumsstelleDto.referat()
+                || studiumsPraktikumsstelle.getDringlichkeit() != praktikumsstelleDto.dringlichkeit()
+                || !nwksAreEqual(nwkMapper.toDto(studiumsPraktikumsstelle.getAssignedNwk()), praktikumsstelleDto.assignedNwk())
+                || !Objects.equals(studiumsPraktikumsstelle.getNamentlicheAnforderung(), praktikumsstelleDto.namentlicheAnforderung())
+                || studiumsPraktikumsstelle.isPlanstelleVorhanden() != praktikumsstelleDto.planstelleVorhanden()
+                || !studiumsPraktikumsstelle.getMeldezeitraumID().toString().equals(praktikumsstelleDto.meldezeitraumID().toString())
+                || !Objects.equals(studiumsPraktikumsstelle.getProgrammierkenntnisse(), praktikumsstelleDto.programmierkenntnisse())
+                || !studiumsPraktikumsstelle.getStudiensemester().equals(praktikumsstelleDto.studiensemester())
+                || studiumsPraktikumsstelle.getStudiengang() != praktikumsstelleDto.studiengang()) {
+            throw new ResourceConflictException("Unerlaubter Versuch der Änderung von Daten");
+        }
+
+        StudiumsPraktikumsstelle updatedPraktikumsstelle = praktikumsstellenMapper.updateStudiumsPraktikumsstelle(studiumsPraktikumsstelle,
+                praktikumsstelleDto);
+        updatedPraktikumsstelle.setDienststelle(normalizeDienststelle(updatedPraktikumsstelle.getDienststelle()));
+
+        studiumsPraktikumsstellenRepository.save(updatedPraktikumsstelle);
+    }
+
+    private boolean nwksAreEqual(NwkDto nwkDto, NwkDto nwkDtoToCompare) {
+        boolean returnValue = nwkDto.active() == nwkDtoToCompare.active();
+
+        if (!nwkDto.id().equals(nwkDtoToCompare.id())) returnValue = false;
+        if (!Objects.equals(nwkDto.vorname(), nwkDtoToCompare.vorname())) returnValue = false;
+        if (!Objects.equals(nwkDto.nachname(), nwkDtoToCompare.nachname())) returnValue = false;
+        if (nwkDto.studiengang() != nwkDtoToCompare.studiengang()) returnValue = false;
+        if (nwkDto.ausbildungsrichtung() != nwkDtoToCompare.ausbildungsrichtung()) returnValue = false;
+        if (!nwkDto.vorlesungstage().equals(nwkDtoToCompare.vorlesungstage()))
+            if (!Objects.equals(nwkDto.jahrgang(), nwkDtoToCompare.jahrgang())) returnValue = false;
+
+        return returnValue;
+    }
```

Comment:
Why you dont use the equal-Method of the record?

Reply:
This method is not anymore required, because the ability to send nwks through the Update....PraktikumsstellenDto is now removed